sktime/transformations/series/scaledlogit.py

```diff
@@ -0,0 +1,164 @@
+    sktime.transformations.series.exponent.SqrtTransformer :
+        Transform input data by taking its square root. Can help compress
+        variance of input series.
+
+    Notes
+    -----
+    | The scaled logit transform is applied if both upper_bound and lower_bound are
+    | not None:
+    |   :math:`log(\frac{x - a}{b - x})`, where a is the lower and b is the upper bound.
+
+    | If upper_bound is None and lower_bound is not the transform applied is:
+    |   :math:`log(x - a)`
+
+    | If lower_bound is None and upper_bound is not the transform applied is:
+    |   :math:`- log(b - x)`
+
+    References
+    ----------
+    .. [1] Hyndsight - Forecasting within limits:
+        https://robjhyndman.com/hyndsight/forecasting-within-limits/
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.transformations.series.scaledlogit import ScaledLogitTransformer
+    >>> from sktime.forecasting.ets import AutoETS
+    >>> from sktime.forecasting.compose import TransformedTargetForecaster
+    >>> y = load_airline()
+    >>> fcaster = TransformedTargetForecaster([
+    ...     ("scaled_logit", ScaledLogitTransformer(0, 650)),
+    ...     ("ets", AutoETS(sp=12, auto=True))
+    ... ])
+    >>> fcaster.fit(y)
+    TransformedTargetForecaster(...)
+    >>> y_pred = fcaster.predict(fh = np.arange(32))
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        # what is the scitype of X: Series, or Panel
+        "scitype:transform-output": "Series",
+        # what scitype is returned: Primitives, Series, Panel
+        "scitype:instancewise": True,  # is this an instance-wise transform?
+        "X_inner_mtype": "np.ndarray",  # which mtypes do _fit/_predict support for X?
+        "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for y?
+        "transform-returns-same-time-index": True,
+        "fit-in-transform": True,
+        "univariate-only": False,
+        "capability:inverse_transform": True,
+        "skip-inverse-transform": False,
+    }
+
+    def __init__(self, lower_bound=None, upper_bound=None):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+        super(ScaledLogitTransformer, self).__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : 2D np.ndarray
+            Data to be transformed
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Ignored argument for interface compatibility
+
+        Returns
+        -------
+        transformed version of X
+        """
+        if self.upper_bound:
+            if np.any(X >= self.upper_bound):
+                raise ValueError("X should not have values greater than upper_bound")
```

Comment:
I would return a warning and return nan instead.

Also, think of a message that is still helpful if this happens in a pipeline. E.g., "X in ScaledLogitTransformer ..."

Reply:
Ok - this will be consistent with the equivalent of passing negative values to the LogTransform. Will change  to warning + NaN.